dags/hindawi/parser.py

```diff
@@ -0,0 +1,200 @@
+        )
+        authors_all + (
+            article.findall(
+                "ns0:metadata/ns1:record/ns0:datafield/[@tag='700']", self.prefixes
+            )
+        )
+        authors = []
+        for author in authors_all:
+            author_ = {
+                "raw_name": author.find(
+                    "ns0:subfield[@code='a']", self.prefixes
+                ).text.replace("\n", ""),
+                "affiliations": self._get_affiliations(author),
+            }
+            orcid = author.find("ns0:subfield[@code='j']", self.prefixes)
+            if orcid.text:
+                author_["orcid"] = orcid.text
+            authors.append(author_)
+
+        return authors
+
+    def _get_affiliations(self, author):
+        affiliations = [
+            affiliation.text.replace("\n", "")
+            for affiliation in author.findall("ns0:subfield[@code='u']", self.prefixes)
+        ]
+        parsed_affiliations = [
+            {
+                "value": affiliation,
+                "organization": (",").join(affiliation.split(",")[:-1]),
+                "country": affiliation.split(", ")[-1:][0],
+            }
+            for affiliation in affiliations
+        ]
+        return parsed_affiliations
+
+    def _get_arxiv(self, article: ET.Element):
+        arxiv = article.find(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='037']/ns0:subfield/[@code='9']",
+            self.prefixes,
+        )
+        if arxiv is None:
+            self.logger.error("there is no Arxiv!")
+            return None
+        if arxiv.text.lower() == "arxiv":
+            arxiv_value = article.find(
+                "ns0:metadata/ns1:record/ns0:datafield/[@tag='037']/ns0:subfield/[@code='a']",
+                self.prefixes,
+            ).text.lower()
+            return [{"value": ("").join(arxiv_value.replace("arxiv:", "").split())}]
+
+    def _get_journal_pages(self, article: ET.Element):
+        journal_pages = article.find(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='773']/ns0:subfield/[@code='c']",
+            self.prefixes,
+        ).text
+        if "-" in journal_pages:
+            return journal_pages.split("-", 1)
+        else:
+            return journal_pages, ""
+
+    def _get_copyright(self, article: ET.Element):
+        copyright_statement = article.find(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='542']/ns0:subfield/[@code='f']",
+            self.prefixes,
+        ).text
+        copyright_year = "".join(i for i in copyright_statement if i.isdigit())
+        return {
+            "copyright_statement": copyright_statement,
+            "copyright_year": copyright_year,
+        }
+
+    def _get_copyright_statement(self, article: ET.Element):
+        return self._get_copyright(article)["copyright_statement"]
+
+    def _get_copyright_year(self, article: ET.Element):
+        return self._get_copyright(article)["copyright_year"]
```

Comment:
if we use it in such way, we run the `_get_copyright` function twice

Reply:
☝️

Reply:
I know, but I don't know the different solution. The last solution, didn't work, because I moved the logic to general parsing, to have this function used once